src/tokens/tokens.service.ts

```diff
@@ -139,55 +139,57 @@ export class TokensService {
+   * expected format (ie incorrect names, missing event subscriptions).
    *
-   * TODO: eventually this migration logic can be pruned
+   * Log a warning if any potential issues are flagged. User may need to delete
+   * subscriptions manually and reactivate the pool directly.
    */
-  async migrate() {
+  async migrationCheck() {
     const name = packStreamName(this.topic, this.instancePath);
     const streams = await this.eventstream.getStreams();
     const existingStream = streams.find(s => s.name === name);
     if (existingStream === undefined) {
       // Look for the old stream name (topic alone)
       const oldStream = streams.find(s => s.name === this.topic);
       if (oldStream !== undefined) {
-        this.logger.warn('Old event stream found - deleting and recreating');
-        await this.eventstream.deleteStream(oldStream.id);
-        await this.init();
+        this.logger.warn('Old event stream found - please delete and recreate!');
+        return true;
       }
-      return;
+      return false;
     }
-    const subscriptions = await this.eventstream.getSubscriptions();
+    const allSubscriptions = await this.eventstream.getSubscriptions();
+    const subscriptions = allSubscriptions.filter(s => s.stream === existingStream.id);
     if (subscriptions.length === 0) {
-      return;
+      return false;
+    }
+
+    const baseSubscription = packSubscriptionName(
+      this.topic,
+      this.instancePath,
+      BASE_SUBSCRIPTION_NAME,
+      tokenCreateEvent,
+    );
+    if (subscriptions.length === 1 && subscriptions[0].name === baseSubscription) {
+      // Special case - only the base subscription exists (with the correct name),
+      // but no pools have been activated. This is ok.
+      return false;
     }
 
     const foundEvents = new Set<string>();
-    for (const sub of subscriptions.filter(s => s.stream === existingStream.id)) {
+    for (const sub of subscriptions) {
       const parts = unpackSubscriptionName(this.topic, sub.name);
       if (parts.event !== undefined && parts.event !== '') {
         foundEvents.add(parts.event);
       }
     }
 
-    if (foundEvents.size === 1 && foundEvents.has(BASE_SUBSCRIPTION_NAME)) {
-      // Special case - only the base subscription exists (with the correct name),
-      // but no pools have been activated. This is ok.
-      return;
-    }
-
-    // Otherwise, expect to have found subscriptions for each of the events.
-    for (const event of ALL_SUBSCRIBED_EVENTS) {
-      if (!foundEvents.has(event)) {
-        this.logger.warn('Incorrect event stream subscriptions found - deleting and recreating');
-        await this.eventstream.deleteStream(existingStream.id);
-        await this.init();
-        return;
-      }
+    // Expect to have found subscriptions for each of the events.
+    if (!ALL_SUBSCRIBED_EVENTS.every(event => foundEvents.has(event))) {
+      this.logger.warn('Incorrect event stream subscriptions found - please delete and recreate!');
```

Comment:
Similar suggestion here.

`The subscriptions under event stream ${existingStream.id} do not include ${ALL_SUBSCRIBED_EVENTS.filter(event => !foundEvents.has(event)).join(",")}. Data related to these events will be not be streamed to FireFly for this token pool`